moveit_simple/include/moveit_simple/joint_lock_options.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2018 Plus One Robotics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JOINT_LOCK_OPTIONS_H
+#define JOINT_LOCK_OPTIONS_H
+
+namespace moveit_simple
+{
+
+typedef unsigned char uchar_t;
+
+enum class JointLockOptions : uchar_t
```

Comment:
What about robots with more than 6 joints?

Reply:
Currently not supported. I'll add some documentation mentioning that.

Reply:
Why not support it? Is it too hard to add?

Reply:
It wouldn't be hard to add, but it would be tedious. 

It is beyond the scope of what we need at the moment. How many joints should we support? up to 7? or infinite?

To "support" it we also need test coverage and we don't have a 7+ DOF robot integrated into testing.

Reply:
That being said, I'd be glad to add it if you would like me to divert time to this task.